src/main/java/org/jboss/modules/JDKSpecific.java

```diff
@@ -185,6 +189,36 @@ static void addInternalPackages(final List<String> list) {
         list.add("jdk.internal.reflect.");
     }
 
+    static <T> Iterable<T> findServices(final Class<T> serviceType, final Predicate<Class<?>> filter, final ClassLoader classLoader) {
+        final Iterator<T> delegate = ServiceLoader.load(serviceType, classLoader).iterator();
+        return new Iterable<T>() {
+            public Iterator<T> iterator() {
+                return new Iterator<T>() {
+                    T next = null;
+
+                    public boolean hasNext() {
+                        T next;
```

Comment:
This method should always return `true` if `this.next` is non-null.  Otherwise `hasNext()` will skip upstream elements.

Reply:
Thanks!